packages/shared_preferences/shared_preferences_android/android/src/main/java/io/flutter/plugins/sharedpreferences/SharedPreferencesPlugin.java

```diff
@@ -133,12 +135,18 @@ public void onDetachedFromEngine(@NonNull FlutterPlugin.FlutterPluginBinding bin
   }
 
   // Gets all shared preferences, filtered to only those set with the given prefix.
+  // Optionally filtered also to only those items in the optional [allowList].
   @SuppressWarnings("unchecked")
-  private @NonNull Map<String, Object> getAllPrefs(@NonNull String prefix) throws RuntimeException {
+  private @NonNull Map<String, Object> getAllPrefs(
+      @NonNull String prefix, @Nullable List<String> allowList) throws RuntimeException {
     Map<String, ?> allPrefs = preferences.getAll();
     Map<String, Object> filteredPrefs = new HashMap<>();
+    Set<String> allowSet = null;
+    if (allowList != null) {
+      allowSet = new HashSet<>(allowList);
+    }
```

Comment:
Optional: consider a ternary:

```
final Set<String> allowSet = allowList == null ? null : new HastSet<>(allowList);
```

Reply:
Since this is a new parem why not pass in a set instead of a list?

Reply:
It's coming from a pigeon api, which doesn't support sets.

Reply:
Though, it could be transformed before being passed into this method.